Pinta.Tools/HistoryItems/GradientHistoryItem.cs

```diff
@@ -0,0 +1,101 @@
+using Cairo;
+using Pinta.Core;
+
+namespace Pinta.Tools;
+
+public class GradientHistoryItem : BaseHistoryItem
+{
+	private readonly GradientTool tool;
+
+	private readonly UserLayer user_layer;
+
+	private readonly SurfaceDiff? user_surface_diff;
+	private ImageSurface? user_surface;
+
+	private GradientData gradient_data;
+
+	public GradientHistoryItem (string icon, string text, ImageSurface passedUserSurface, UserLayer passedUserLayer, GradientData passedData, GradientTool passedTool) : base (icon, text)
```

Comment:
Since some of this code is the same as the original SimpleHistoryItem used in the old version of the tool, I think it might be possible to avoid that duplicate code:
- You could have the gradient history item have a SimpleHistoryItem internally that it delegates to, and then you only need to implement the extra stuff specific to the gradient tool
- Or, a CompoundHistoryItem could be used to combine the two